proversity_reports_script/report_backend/enrollment_report.py

```diff
@@ -0,0 +1,280 @@
+            """
+            Takes the argument full_name and returns the first name and last name separately.
+            If the full name value cannot be splitted at least once, it will return
+            the full_name argument as name and last name.
+
+            Args:
+                full_name: The student full name.
+            Returns:
+                name: The name of the student.
+                last_name: The last name of the student.
+            """
+            result = full_name.split(' ', 1)
+
+            if len(result) == 2:
+                return result
+
+            return full_name, full_name
+
+        salesforce_data = []
+        batch = []
+
+        for index, user in enumerate(course_data, 1):
+            contact_id = user.get('contact_id', '')
+
+            # If the user doesn't have a user_id, the creation cannot be possible.
+            if not user.get('user_id', ''):
+                continue
+
+            # If the user has already a contact id,
+            # it's not neccesary to request the creation again.
+            if contact_id and index == len(course_data):
+                salesforce_data.append(batch)
+                batch = []
+                continue
+            elif contact_id:
+                continue
+
+            first_name, last_name = get_first_and_last_name(user.get('full_name', ''))
+            username = user.get('username', '')
+
+            user_salesforce_data = {
+                'FirstName': first_name if first_name else username,
+                'LastName': last_name if last_name else username,
+                'Email': user.get('email', ''),
+                'Company': self.extra_data.get('COMPANY_NAME', ''),
+                'Institution_Hidden': self.extra_data.get('INSTITUTION_HIDDEN_PREFIX', ''),
+                'Type_Hidden': self.extra_data.get('TYPE_HIDDEN', ''),
+                'Program_of_Interest': self.extra_data.get('PROGRAM_OF_INTEREST', ''),
+                'Intake_of_Intent': user.get('intake_of_intent', ''),
+                'Lead_Source': DEFAULT_LEAD_SOURCE,
+                'Secondary_Source': '',
+                'Tertiary_Source': '',
+                'Program_Code': course,
+                'Drupal_ID': user.get('user_id', ''),
+            }
+
+            batch.append(user_salesforce_data)
+
+            if len(batch) == BATCH_SIZE or index == len(course_data):
+                salesforce_data.append(batch)
+                batch = []
+
+        return salesforce_data
+
+    def _create_saleforce_enrollments(self, enrollment_data):
+        """
+        Creates the enrollment list into Salesforce.
+
+        Args:
+            enrollment_data: List of list with the enrollment data.
+        Returns:
+            salesforce_result: List wiht the contact id data from Salesforce.
+        """
+        request_session = requests.Session()
+        salesforce_details = self._get_salesforce_details()
+        api_url = '{}{}'.format(
```

Comment:
Add validation for self.extra_data.get('SALESFORCE', {}).get('SALESFORCE_API_ENROLLMENT_URL', '')

Reply:
Addressed.